app/controllers/posts_controller.rb

```diff
@@ -2,22 +2,29 @@ class PostsController < ApplicationController
   # before_action :check_logged_in, only: [:create]
   before_action :location, only: [:index, :show]
   respond_to :html, :js
+  POSTS_PER_PAGE = 25 # tried 50 and it's way too long imo
+  COMMENTS_PER_PAGE = 50
 
   def index
+    visible_posts = Post.within_location(@location)
     @sort = params[:sort]
+    @posts_page = (params[:posts_page] || 1).to_i
+    @posts_page_max = (visible_posts.size + (POSTS_PER_PAGE - 1)) / POSTS_PER_PAGE
     @posts = case @sort
              when 'spiciest'
-               Post.within_location(@location).order('like_count DESC')
+              visible_posts.order('like_count DESC').limit(POSTS_PER_PAGE).offset((@posts_page - 1) * POSTS_PER_PAGE)
```

Comment:
you can probably define paginate(@posts_page) as `.limit().offset()` to avoid cope duplication